skrub/_squashing_scaler.py

```diff
@@ -0,0 +1,341 @@
+import numbers
+
+import numpy as np
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.utils.validation import check_is_fitted
+
+from . import _dataframe as sbd
+from ._on_each_column import OnEachColumn, RejectColumn, SingleColumnTransformer
+
+
+class SingleColumnSquashingScaler(SingleColumnTransformer):
```

Comment:
I seems to me that the logic could be implemented well at the level of multiple columns. Nan handling can be done using np.nanmean and np.nanquantile.

I think that this would be beneficial